cdap-cli/src/main/java/co/cask/cdap/cli/util/ArgumentParser.java

```diff
@@ -61,6 +61,30 @@
   }
 
   /**
+   * Parses a map in the format: "key1=a key2=b .."
+   *
+   * @param mapString {@link String} representation of the map
+   * @return the map
+   */
+  public static Map<String, Integer> parseStringIntegerMap(String mapString) {
+    if (mapString == null || mapString.isEmpty()) {
+      return ImmutableMap.of();
+    }
+
+    ImmutableMap.Builder<String, Integer> result = ImmutableMap.builder();
```

Comment:
Yeah, I also want to do that, but how to construct a Map with Generic type and convert String to the corresponding Generic type?